### PR TITLE
Adherence statistics (BRIDGE-3212)

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java [hunk 1]
```diff
@@ -488,4 +494,25 @@ protected WeeklyAdherenceReport deriveWeeklyAdherenceFromStudyAdherenceReport(St
...
     }
+    
+    public AdherenceStatistics getAdherenceStatistics(String appId, String studyId, Integer adherenceThreshold) {
+        checkNotNull(appId);
+        checkNotNull(studyId);
+        
+        Study study = studyService.getStudy(appId, studyId, true);
+        
+        scheduleService.getScheduleForStudy(appId, study)
+            .orElseThrow(() -> new EntityNotFoundException(Schedule2.class));
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java:505]: Elsewhere we report no schedule if there's no schedule...do we want to do the same here?

File: src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java [hunk 2]
```diff
@@ -488,4 +494,25 @@ protected WeeklyAdherenceReport deriveWeeklyAdherenceFromStudyAdherenceReport(St
...
+        
+        if (adherenceThreshold == null) {
+            adherenceThreshold = study.getAdherenceThresholdPercentage();
+        }
+        if (adherenceThreshold == null) {
+            throw new BadRequestException(NO_THRESHOLD_VALUE_ERROR);
+        }
+        if (adherenceThreshold < 1 || adherenceThreshold > 100) {
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java:513]: Less than zero.

File: src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java:505]: Leave this, it's correct, and filed BRIDGE-3273 to change other adherence APIs to behave in a similar manner.

File: src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/AdherenceStatistics.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package org.sagebionetworks.bridge.models.schedules2.adherence;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.google.common.collect.ImmutableMap;
```
Comment (jnelsonjava) [inline src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/AdherenceStatistics.java]: Might be an unused import.

**Review by jnelsonjava — APPROVED** [review]
Looks good, just an import line to check.

File: src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/AdherenceStatistics.java (hunk 3 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/AdherenceStatistics.java]: removed (along with 2 others)

**Pull request merged**